.github/workflows/sketcher-builder.yml

```diff
@@ -0,0 +1,113 @@
+name: Sketcher Build and Test
+
+on:
+  workflow_dispatch: ~
+  push: ~
+  pull_request: ~
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  # TODO: clang-format hook
+  build_test_sketcher:
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+        - {
+            name: "macOS",
+            os: macos-latest,
+            app: "build/sketcher_app",
+          }
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Read external/versions.json
+        shell: bash
+        run: |
+          {
+              echo 'VERSIONS_JSON<<EOF'
+              cat external/versions.json
+              echo -e '\nEOF'
+          } >> "$GITHUB_ENV"
+
+      - name: CPU count
+        shell: python
```

Comment:
If you have python, then you might want to install the "ninja" package and pass "-G Ninja" to CMake. This will make builds faster on windows, as MSBuild cannot build in parallel processes.

Reply:
Will do.